content/docs/error-boundaries.md

```diff
@@ -0,0 +1,137 @@
+
+#### componentDidCatch Parameters
+
+`error` is an error that has been thrown.
+
+`info` is an object with `componentStack` key. The property has information about component stack during thrown error.
+
+```js
+//...
+componentDidCatch(error, info) {
+  
+  /* Example stack information:
+     in ComponentThatThrows (created by App)
+     in ErrorBoundary (created by App)
+     in div (created by App)
+     in App
+  */
+  logComponentStackToMyService(info.componentStack);
+}
+
+//...
+```
+
+## Live Demo
+
+Check out [this example of declaring and using an error boundary](https://codepen.io/gaearon/pen/wqvxGa?editors=0010) with [React 16 beta](https://github.com/facebook/react/issues/10294).
+
+
+## Where to Place Error Boundaries
+
+The granularity of error boundaries is up to you. You may wrap top-level route components to display a “Something went wrong” message to the user, just like server-side frameworks often handle crashes. You may also wrap individual widgets in an error boundary to protect them from crashing the rest of the application.
+
+
+## New Behavior for Uncaught Errors
+
+This change has an important implication. **As of React 16, errors that were not caught by any error boundary will result in unmounting of the whole React component tree.**
+
+We debated this decision, but in our experience it is worse to leave corrupted UI in place than to completely remove it. For example, in a product like Messenger leaving the broken UI visible could lead to somebody sending a message to the wrong person. Similarly, it is worse for a payments app to display a wrong amount than to render nothing.
+
+This change means that as you migrate to React 16, you will likely uncover existing crashes in your application that have been unnoticed before. Adding error boundaries lets you provide better user experience when something goes wrong.
+
+For example, Facebook Messenger wraps content of the sidebar, the info panel, the conversation log, and the message input into separate error boundaries. If some component in one of these UI areas crashes, the rest of them remain interactive.
+
+We also encourage you to use JS error reporting services (or build your own) so that you can learn about unhandled exceptions as they happen in production, and fix them.
+
+
+## Component Stack Traces
+
+React 16 prints all errors that occurred during rendering to the console in development, even if the application accidentally swallows them. In addition to the error message and the JavaScript stack, it also provides component stack traces. Now you can see where exactly in the component tree the failure has happened:
+
+<img src="../images/docs/error-boundaries-error-log.png" style="max-width:100%" alt="Error caught by Error Boundary component">
```

Comment:
It might be nice to share these images between the blog post and the docs page, since they're the same. Maybe `git mv` the images into `images/docs` and then update the blog markdown to reference them there?